api/post.go

```diff
@@ -663,38 +663,43 @@ func sendNotifications(c *Context, post *model.Post, team *model.Team, channel *
 					sessions := result.Data.([]*model.Session)
 					alreadySeen := make(map[string]string)
 
-					for _, session := range sessions {
-						if len(session.DeviceId) > 0 && alreadySeen[session.DeviceId] == "" &&
-							(strings.HasPrefix(session.DeviceId, model.PUSH_NOTIFY_APPLE+":") || strings.HasPrefix(session.DeviceId, model.PUSH_NOTIFY_ANDROID+":")) {
-							alreadySeen[session.DeviceId] = session.DeviceId
-
-							msg := model.PushNotification{}
-							msg.Badge = 1
-							msg.ServerId = utils.CfgDiagnosticId
-
-							if strings.HasPrefix(session.DeviceId, model.PUSH_NOTIFY_APPLE+":") {
-								msg.Platform = model.PUSH_NOTIFY_APPLE
-								msg.DeviceId = strings.TrimPrefix(session.DeviceId, model.PUSH_NOTIFY_APPLE+":")
-							} else if strings.HasPrefix(session.DeviceId, model.PUSH_NOTIFY_ANDROID+":") {
-								msg.Platform = model.PUSH_NOTIFY_ANDROID
-								msg.DeviceId = strings.TrimPrefix(session.DeviceId, model.PUSH_NOTIFY_ANDROID+":")
-							}
+					pushServer := *utils.Cfg.EmailSettings.PushNotificationServer
+					if pushServer == model.MHPNS && (!utils.IsLicensed || !*utils.License.Features.MHPNS) {
+						l4g.Warn(utils.T("api.post.send_notifications_and_forget.push_notification.mhpnsWarn"))
+					} else {
+						for _, session := range sessions {
+							if len(session.DeviceId) > 0 && alreadySeen[session.DeviceId] == "" &&
+								(strings.HasPrefix(session.DeviceId, model.PUSH_NOTIFY_APPLE+":") || strings.HasPrefix(session.DeviceId, model.PUSH_NOTIFY_ANDROID+":")) {
+								alreadySeen[session.DeviceId] = session.DeviceId
+
+								msg := model.PushNotification{}
+								msg.Badge = 1
+								msg.ServerId = utils.CfgDiagnosticId
+
+								if strings.HasPrefix(session.DeviceId, model.PUSH_NOTIFY_APPLE+":") {
+									msg.Platform = model.PUSH_NOTIFY_APPLE
+									msg.DeviceId = strings.TrimPrefix(session.DeviceId, model.PUSH_NOTIFY_APPLE+":")
+								} else if strings.HasPrefix(session.DeviceId, model.PUSH_NOTIFY_ANDROID+":") {
+									msg.Platform = model.PUSH_NOTIFY_ANDROID
+									msg.DeviceId = strings.TrimPrefix(session.DeviceId, model.PUSH_NOTIFY_ANDROID+":")
+								}
 
-							if channel.Type == model.CHANNEL_DIRECT {
-								msg.Message = senderName + userLocale("api.post.send_notifications_and_forget.push_message")
-							} else {
-								msg.Message = senderName + userLocale("api.post.send_notifications_and_forget.push_mention") + channelName
-							}
+								if channel.Type == model.CHANNEL_DIRECT {
+									msg.Message = senderName + userLocale("api.post.send_notifications_and_forget.push_message")
+								} else {
+									msg.Message = senderName + userLocale("api.post.send_notifications_and_forget.push_mention") + channelName
+								}
```

Comment:
It wasn't removed, that just came in after 2.1 and these changes are based off the 2.1 release branch, not master